detox/android/detox/src/main/java/com/wix/detox/espresso/DetoxAction.java

```diff
@@ -168,67 +169,24 @@ public void perform(UiController uiController, View view) {
         });
     }
 
-    private final static float EDGE_FUZZ_FACTOR = 0.083f;
-
     /**
      * Swipes the View in a direction.
      *
      * @param direction Direction to swipe (see {@link MotionDir})
      * @param fast true if fast, false if slow
-     *
+     * @param amount relative value from 0 to 1
+     * @param startOffsetPercentX relative value from 0 to 1
+     * @param startOffsetPercentY relative value from 0 to 1
      */
-    public static ViewAction swipeInDirection(final int direction, boolean fast) {
-        if (fast) {
-            switch (direction) {
-                case MOTION_DIR_LEFT:
-                    return swipeLeft();
-                case MOTION_DIR_RIGHT:
-                    return swipeRight();
-                case MOTION_DIR_UP:
-                    return swipeUp();
-                case MOTION_DIR_DOWN:
-                    return swipeDown();
-                default:
-                    throw new RuntimeException("Unsupported swipe direction: " + direction);
-            }
-        }
-
-        switch (direction) {
-            case MOTION_DIR_LEFT:
-                return actionWithAssertions(new GeneralSwipeAction(Swipe.SLOW,
-                        translate(GeneralLocation.CENTER_RIGHT, -EDGE_FUZZ_FACTOR, 0),
-                        GeneralLocation.CENTER_LEFT, Press.FINGER));
-            case MOTION_DIR_RIGHT:
-                return actionWithAssertions(new GeneralSwipeAction(Swipe.SLOW,
-                        translate(GeneralLocation.CENTER_LEFT, EDGE_FUZZ_FACTOR, 0),
-                        GeneralLocation.CENTER_RIGHT, Press.FINGER));
-            case MOTION_DIR_UP:
-                return actionWithAssertions(new GeneralSwipeAction(Swipe.SLOW,
-                        translate(GeneralLocation.BOTTOM_CENTER, 0, -EDGE_FUZZ_FACTOR),
-                        GeneralLocation.TOP_CENTER, Press.FINGER));
-            case MOTION_DIR_DOWN:
-                return actionWithAssertions(new GeneralSwipeAction(Swipe.SLOW,
-                        translate(GeneralLocation.TOP_CENTER, 0, EDGE_FUZZ_FACTOR),
-                        GeneralLocation.BOTTOM_CENTER, Press.FINGER));
-            default:
-                throw new RuntimeException("Unsupported swipe direction: " + direction);
-        }
+    public static ViewAction swipeInDirection(final int direction, boolean fast, double amount, double startOffsetPercentX, double startOffsetPercentY) {
+        final boolean isInCompatibilityMode = (fast && amount == Double.NaN && startOffsetPercentX == Double.NaN && startOffsetPercentY == Double.NaN);
```

Comment:
TBH, I was not familiar with the Java-ish representation of `NaN`. Are you 100% that this corresponds to whichever value that is provided by JS when offsets are not provided?

Reply:
btw, this what android studio has to say about this
1. `isInCompatibilityMode` is always `false` (and why? -)
2. 
```
Comparison to 'Double.NaN' is always false 
 Inspection info: Reports any equality or inequality comparisons to Double.NaN or Float.NaN. Equality comparison to these values is always false. Instead, use the Double.isNaN() of Float.isNaN() methods instead.
```

Reply:
JS provides a `null`, because a JSON cannot hold `NaN` values. 🤦‍♂️ So the JSON parser needs to be aware of this, and put `NaN` where appropriate, if the JS didn't provide a value.

Reply:
![image](https://user-images.githubusercontent.com/1962469/99144122-baaa2100-266b-11eb-9779-a636107c788e.png)

@d4vidi , it works the same in Java.